File: src/main/java/dev/obscuria/tooltips/client/tooltip/element/slot/BlankSlot.java
package dev.obscuria.tooltips.client.tooltip.element.slot;

import com.github.bsideup.jabel.Desugar;
import dev.obscuria.tooltips.client.render.GuiGraphics;

@Desugar
public record BlankSlot() implements TooltipSlot {

    public static final BlankSlot INSTANCE = new BlankSlot();

    @Override
    public void render(GuiGraphics graphics, int x, int y, int width, int height) {}
}

File: common/src/main/kotlin/dev/obscuria/tooltips/client/style/slot/BlankSlot.kt
package dev.obscuria.tooltips.client.style.slot

import com.mojang.serialization.Codec
import net.minecraft.client.gui.GuiGraphics

object BlankSlot : ITooltipSlot {

    val CODEC: Codec<BlankSlot> = Codec.unit(BlankSlot)

    override fun codec(): Codec<BlankSlot> = CODEC

    override fun render(graphics: GuiGraphics, x: Int, y: Int, width: Int, height: Int) {}
}